client/ide/lib/views/contentsearch/idecontentsearchview.coffee

```diff
@@ -0,0 +1,292 @@
+        return
+
+      targetIndex = if direction is 'up' then currentIndex - 1 else currentIndex + 1
+      nextTerm    = previousTerms[targetIndex]
+
+      if not nextTerm and direction is 'down'
+        return @findInput.setValue ''
+
+      @setSearchText nextTerm
+
+
+  setSearchText: (text) ->
+
+    @findInput.setValue text
+
+    kd.utils.defer =>
+      kd.utils.moveCaretToEnd @findInput.getElement()
+
+
+  search: ->
+
+    searchText = Encoder.XSSEncode @findInput.getValue()
+
+    return @searchButton.hideLoader()  if searchText is ''
+
+    @searchButton.showLoader()
+    @searchText     = searchText
+    @rootPath       = Encoder.XSSEncode @whereInput.getValue()
+    @selections     = @choices.getValue()
+
+    isCaseSensitive = @selections.indexOf('case-sensitive') > -1
+    isWholeWord     = @selections.indexOf('whole-word') > -1
+    isRegExp        = @selections.indexOf('regex') > -1
+    exts            = editorSettings.getAllExts()
+    include         = "\\*{#{exts.join ','}}"
+    exclureDirs     = Object.keys editorSettings.ignoreDirectories
+    exclureDirs     = " --exclude-dir=#{exclureDirs.join ' --exclude-dir='}"
+    searchText      = @searchText
+
+    unless isRegExp
+      splitText     = searchText.split '\\n'
+      splitText     = splitText.map @grepEscapeRegExp
+      searchText    = splitText.join '\\n'
+
+    searchText      = searchText.replace (new RegExp "\\\'", 'g'), "'\\''"
+    searchText      = searchText.replace /-/g, '\\-'
+
+    flags = [
+      '-s'                           # Silent mode
+      '-r'                           # Recursively search subdirectories listed.
+      '-n'                           # Each output line is preceded by its relative line number in the file
+      '-A 3'                         # Print num lines of trailing context after each match.
+      '-B 3'                         # Print num lines of trailing context before each match.
+      '-i'                           # Match case insensitively
+      '-w'                           # Only match whole words
+      '--color=never'                # Disable color output to get plain text
+      '--binary-files=without-match' # Do not search binary files
+    ]
+
+    flags.splice flags.indexOf('-i'), 1  if isCaseSensitive
+    flags.splice flags.indexOf('-w'), 1  unless isWholeWord
+
+    command = "grep #{flags.join ' '} #{exclureDirs} --include=#{include} '#{searchText}' \"#{@escapeShell @rootPath}\""
+
+    { machine } = @getData()
+    machine.getBaseKite().exec({ command })
+      .then  (res) => @formatOutput res.stdout, @bound 'createResultsView'
+      .catch (err) =>
+        @showWarning 'Something went wrong, please try again.'
+        kd.warn err
+
+    terms = @localStorageController.getValue('PreviousSearchTerms') or []
+
+    terms.push searchText
+    @localStorageController.setValue 'PreviousSearchTerms', terms
+
+
+  escapeRegExp: (str) -> str.replace /([.*+?\^${}()|\[\]\/\\])/g, '\\$1'
```

Comment:
keep all the regular expressions in a constant on top of the file above class description.

e.g.

```coffee
REGEXES =
  escapeRegExp: /([.*+?\^${}()|\[\]\/\\])/g
  escapeShell: /([\\"'`$\s\(\)<>])/g
…
```